include/experimental/__p1673_bits/blas1_givens.hpp

```diff
@@ -344,12 +344,20 @@ void givens_rotation_setup(const complex<Real>& f,
   }
 }
 
-template<class inout_vector_1_t,
-         class inout_vector_2_t,
+// FIXME (Hoemmen 2021/05/28) Latest version of P0009 (mdspan) uses size_t
+// instead of ptrdiff_t, but the implementation hasn't changed yet.
+template<class ElementType1,
+         ptrdiff_t ext1,
+         class Layout1,
+         class Accessor1,
+         class ElementType2,
+         ptrdiff_t ext2,
+         class Layout2,
+         class Accessor2,
          class Real>
 void givens_rotation_apply(
-  inout_vector_1_t x,
-  inout_vector_2_t y,
+  std::experimental::basic_mdspan<ElementType1, std::experimental::extents<ext1>, Layout1, Accessor1> x,
+  std::experimental::basic_mdspan<ElementType2, std::experimental::extents<ext2>, Layout2, Accessor2> y,
   const Real c,
```

Comment:
It occurs to me that the proposal constrains Givens rotations to work only on floating-point types, but we can always go back to fix that later.  (The relevant concept is `std::floating_point`, so we could say `std::floating_point Real` instead of `class Real`.)

Reply:
BRB, installing gcc 10 so I can actually `#include<concepts>`